apps/prairielearn/src/pages/instructorAssessmentGroups/instructorAssessmentGroups.html.ts

```diff
@@ -0,0 +1,557 @@
+    </form>
+  `;
+}
+
+function RemoveMembersForm({ row, csrfToken }: { row: GroupUsersRow; csrfToken: string }) {
+  return html`
+    <form name="delete-member-form" method="POST">
+      <div class="form-group">
+        <label for="delete-member-form-${row.group_id}">UID:</label>
+        <select class="form-control" name="user_id" id="delete-member-form-${row.group_id}">
+          ${row.users.map((user) => {
+            return html` <option value="${user.user_id}">${user.uid}</option> `;
+          })}
+        </select>
+      </div>
+      <input type="hidden" name="__action" value="delete_member" />
+      <input type="hidden" name="__csrf_token" value="${csrfToken}" />
+      <input type="hidden" name="group_id" value="${row.group_id}" />
+      <button
+        type="button"
+        class="btn btn-secondary"
+        onclick="$('#row${row.group_id}Popoverdeletemember').popover('hide')"
+      >
+        Cancel
+      </button>
+      <button type="submit" class="btn btn-danger" ${row.users.length > 0 ? '' : 'disabled'}>
+        Delete
+      </button>
+    </form>
+  `;
+}
+
+function UploadAssessmentGroupsModal({ csrfToken }: { csrfToken: string }) {
+  return Modal({
+    id: 'uploadAssessmentGroupsModal',
+    title: 'Upload new group assignments',
+    formEncType: 'multipart/form-data',
+    body: html`
+      <p>Upload a CSV file in the format of:</p>
+      <code class="mb-0 text-dark">
+        groupName,UID<br />
+        groupA,one@example.com<br />
+        groupA,two@example.com<br />
+        groupB,three@example.com<br />
+        groupB,four@example.com</code
+      >
+      <!-- Closing code tag not on its own line to improve copy/paste formatting -->
+      </div>
+      <div class="modal-body">
+        <p>
+          The <code>groupname</code> column should be a unique identifier for each group. To
+          change the grouping, link uids to the groupname.
+        </p>
+          <div class="form-group">
+            <div class="custom-file">
+              <input
+                type="file"
+                accept=".csv"
+                name="file"
+                class="custom-file-input"
+                id="uploadAssessmentGroupsFileInput"
+              />
+              <label class="custom-file-label" for="uploadAssessmentGroupsFileInput">
+                Choose CSV file
+              </label>
+            </div>
+          </div>
+          <div class="d-flex justify-content-end">
+            <div class="form-group mb-0">
+              <input type="hidden" name="__action" value="upload_assessment_groups" />
+              <input type="hidden" name="__csrf_token" value="${csrfToken}" />
+              <button type="button" class="btn btn-secondary" data-dismiss="modal">
+                Cancel
+              </button>
+              <button type="submit" class="btn btn-primary">Upload</button>
+            </div>
+          </div>
```

Comment:
I'd typically expect this stuff to be explicitly in a modal footer (the `footer` property of our `Modal` options).

Reply:
Moved to the `footer` and removed unnecessary divs in [49b24ad](https://github.com/PrairieLearn/PrairieLearn/pull/9550/commits/49b24ad27a42ff43993f2210f2ac2a3d611c3f11).